config/application.rb

```diff
@@ -12,24 +13,16 @@ class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 7.0
 
-    # Settings in config/environments/* take precedence over those specified here.
-    # Application configuration should go into files in config/initializers
-    # -- all .rb files in that directory are automatically loaded.
-
-    # Custom directories with classes and modules you want to be autoloadable.
-    # config.autoload_paths += %W(#{config.root}/extras)
-    # config.autoload_paths += config.root + File.join('app', 'legacy_lib')
-    # config.eager_load_paths += Dir["#{config.root}/lib/**/"]
-
-    # config.paths.add File.join('lib'), glob: File.join('**')
-
-    # Only load the plugins named here, in the order given (default is alphabetical).
-    # :all can be used as a placeholder for all plugins not explicitly named.
-    # config.plugins = [ :exception_notification, :ssl_requirement, :all ]
-
-    # Activate observers that should always be running.
-    # config.active_record.observers = :cacher, :garbage_collector, :forum_observer
+    # Please, add to the `ignore` list any other `lib` subdirectories that do
+    # not contain `.rb` files, or that should not be reloaded or eager loaded.
+    # Common ones are `templates`, `generators`, or `middleware`, for example.
+    config.autoload_lib(ignore: %w[assets core_ext generators tasks])
```

Comment:
The `lib` directory is now autoloaded in Rails 7.1 but we don't necessarily want that for everything, especially cases where we're extending core classes. This handles that situation.